content/en/security_platform/application_security/troubleshooting.md

```diff
@@ -0,0 +1,437 @@
+- spring-web
+- spring-webflux
+
+{{< /programming-lang >}}
+{{< programming-lang lang=".NET" >}}
+
+For .Net, the required integrations are the following:
+
+- AspNet
+
+**Note:** If AspNetCore is disabled, Application Security should still work with this framework.
+
+{{< /programming-lang >}}
+{{< programming-lang lang="PHP" >}}
+
+**[NEED INFO]**
+
+{{< /programming-lang >}}
+{{< programming-lang lang="go" >}}
+
+For Go, the required integrations are the following:
+
+- net/http
+- Chi
+- Echo
+- Gorilla/mux
+- gRPC
+
+**Note:** They can’t be deactivated but rather just not integrated into the program’s source code.
+
+{{< /programming-lang >}}
+{{< programming-lang lang="NodeJS" >}}
+
+For NodeJS, the required integrations are the following:
+
+- HTTP
+
+{{< /programming-lang >}}
+{{< programming-lang lang="ruby" >}}
+
+For Ruby, the required integrations are the following:
+
+- Rack
+
+**Note:** Rack can be manually added or automatically added via Rails or Sinatra integration. If manually added, the tracer middleware must appear before the security middleware in the Rack stack.
+
+{{< /programming-lang >}}
+{{< /programming-lang-wrapper >}}
+
+### 3. Application to Datadog Agent
+
+ To troubleshoot this step of the process, do the following: 
+
+- Check the details of the running Agent at this address `http://<agent-machine-name>:<agent-port>/info`, usually `http://localhost:8126/info`. 
+- Ensure there are no transmission errors related to spans in the [tracer logs][4]. 
+- If the agent is installed on a separate machine, check that `DD_AGENT_HOST` and optionally `DD_TRACE_AGENT_PORT` are set, or that `DD_TRACE_AGENT_URL` is set for the application / tracing library.
+
+### 4. Datadog Agent to backend
+
+Application Security events are sent over [spans][5]. To confirm that spans are successfully transmitted to the backend, check that the tracer logs contain logs like this:
+
+```
+2021-11-29 21:19:58 CET | TRACE | INFO | (pkg/trace/info/stats.go:111 in LogStats) | [lang:.NET lang_version:5.0.10 interpreter:.NET tracer_version:1.30.1.0 endpoint_version:v0.4] -> traces received: 2, traces filtered: 0, traces amount: 1230 bytes, events extracted: 0, events sampled: 0
+```
+
+If spans are not being transmitted, then the tracer logs will contain logs like this:
+
+```
+2021-11-29 21:18:48 CET | TRACE | INFO | (pkg/trace/info/stats.go:104 in LogStats) | No data received
+```
+
+Below are additional troubleshooting steps for specific languages.
+
+{{< programming-lang-wrapper langs="java,.NET,go,ruby,PHP,NodeJS" >}}
+{{< programming-lang lang="java" >}}
+The Java library uses [SLF4J][1] for logging. Add the following runtime flags so that the tracer logs to a file:
+
+```java
+ -Ddatadog.slf4j.simpleLogger.defaultLogLevel=info
+ -Ddatadog.slf4j.simpleLogger.logFile=dd.log
+```
+
+After the service starts, the tracer will log to the specified file. Datadog recommends using `INFO` for the log level because `DEBUG` logs are verbose.
```

Comment:
```suggestion
After the service starts, the tracer logs to the specified file. Datadog recommends using `INFO` for the log level because `DEBUG` logs are verbose.
```